### PR TITLE
RDoc-1697 java index query code samples

### PR DESCRIPTION
**Pull request by morhilai**
In both Java and Csharp, there were problems in the code samples for index definitions:
- Java indexes had nested `Result` classes that were not necessary (they were probably copied from the Csharp code where they are necessary)
- Csharp indexes did not use the Result class in the map function

### CODE REVIEW

File: Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/MapReduceIndexes.java [hunk 1]
```diff
@@ -235,26 +106,26 @@ public MapReduceIndexes() {
         try (IDocumentStore store = new DocumentStore()) {
             try (IDocumentSession session = store.openSession()) {
                 //region map_reduce_0_1
-                List<Products_ByCategory.Result> results = session
-                    .query(Products_ByCategory.Result.class, Products_ByCategory.class)
+                List<Product> results = session
```
Comment (ml054) [inline Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/MapReduceIndexes.java]: it is map reduce so I think it isn't correct.

File: Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/MapReduceIndexes.java [hunk 2]
```diff
@@ -235,26 +106,26 @@ public MapReduceIndexes() {
...
                     .whereEquals("Category", "Seafood")
                     .toList();
                 //endregion
             }
 
             try (IDocumentSession session = store.openSession()) {
                 //region map_reduce_1_1
-                List<Products_Average_ByCategory.Result> results = session
-                    .query(Products_Average_ByCategory.Result.class, Products_Average_ByCategory.class)
+                List<Product> results = session
```
Comment (ml054) [inline Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/MapReduceIndexes.java]: same here

File: Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/MapReduceIndexes.java [hunk 3]
```diff
@@ -235,26 +106,26 @@ public MapReduceIndexes() {
...
                     .whereEquals("Category", "Seafood")
                     .toList();
                 //endregion
             }
 
             try (IDocumentSession session = store.openSession()) {
                 //region map_reduce_2_1
-                List<Product_Sales.Result> results = session
-                    .query(Product_Sales.Result.class, Product_Sales.class)
+                List<Product> results = session
```
Comment (ml054) [inline Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/MapReduceIndexes.java]: same here

File: Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/Querying/Highlights.java [hunk 4]
```diff
@@ -127,9 +106,9 @@ public Highlights() {
                 highlightingOptions.setPreTags(new String[] { "**" });
                 highlightingOptions.setPostTags(new String[] { "**" });
                 highlightingOptions.setGroupKey("category");
-                List<BlogPosts_ByCategory_Content.Result> results = session
+                List<BlogPost> results = session
```
Comment (ml054) [inline Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/Querying/Highlights.java:109]: I not sure it is correct. Can we skype about that?

Comment (morhilai) [reply Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/Querying/Highlights.java]: Fixed

**Pull request merged**